workshops/util.py

```diff
@@ -137,47 +137,29 @@ def verify_upload_person_task(data):
         email = item.get('email', None)
         personal = item.get('personal', None)
         family = item.get('family', None)
+        person_id = item.get('existing_person_id', None)
         person = None
 
-        if email:
+        if person_id:
             try:
-                # check if first and last name matches person in the database
-                person = Person.objects.get(email__iexact=email)
-
-                for which, actual, uploaded in (
-                        ('personal', person.personal, personal),
-                        ('family', person.family, family)
-                ):
-                    if (actual == uploaded) or (not actual and not uploaded):
-                        pass
-                    else:
-                        errors.append('{0} mismatch: database "{1}" '
-                                      'vs uploaded "{2}".'
-                                      .format(which, actual, uploaded))
-
-            except Person.DoesNotExist:
-                # in this case we need to add a new person
-                pass
-
+                person = Person.objects.get(id=int(person_id))
+            except (ValueError, TypeError, Person.DoesNotExist):
+                person = None
+                info.append('Could not match selected person. New record will '
+                            'be created.')
             else:
-                if existing_event and person and existing_role:
-                    # person, their role and a corresponding event exist, so
-                    # let's check if the task exists
-                    try:
-                        Task.objects.get(event=existing_event, person=person,
-                                         role=existing_role)
-                    except Task.DoesNotExist:
-                        info.append('Task will be created.')
-                    else:
-                        info.append('Task already exists.')
-        else:
+                info.append('Existing record for person will be used.')
+
+        if not email and not person:
             info.append('It\'s highly recommended to add an email address.')
 
         if person:
-            # force username from existing record
+            # force details from existing record
+            item['personal'] = person.personal
+            item['family'] = person.family
+            item['email'] = person.email
```

Comment:
In lines 192-194 you make validation using `personal`, but here you use `person.personal`. So when I click "use that person", and then change personal name, and then click "verify", then: 1) my changes are ignored 2) the changed value is validated instead of the one that is actually used (that is, the one from db). The same applies to family and email.